src/decide.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+import "google/protobuf/empty.proto";
+
+package decide;
+
+/* The payload for a requested state change to a component. Component state may
+   be a scalar (int32) or a label (string) */
+message StateChange {
+  string component = 1;
+  oneof state {
+    int32 scalar = 4;
+    string label = 5;
+  }
+}
+
+/* The payload for a requested change to the parameters for a component */
+message ComponentParams {
+  string component = 1;
+  // The documentation for the component must define how the parameter value will be parsed from the string.
+  map<string, string> parameters = 2;
+}
+
+/* This is what gets sent on the wire */
+message Request {
+  string protocol = 1;
+  oneof request {
+    // request change to state of a component. Reply type is ok or error.
+    StateChange state_change = 2;
+    // request reset to default state; value is name of component. Reply type is ok or error
+    string state_reset = 3;
+    // request lock on experiment; value is name of experiment. Reply type is ok or error
+    string lock_expt = 16;
```

Comment:
How are we enforcing locks? I was originally thinking that decide-ctrl would return an authorization token in reply to `lock_expt` that would have to be included in all future requests (except for `force_unlock_expt` for cases when the experiment has crashed)

Reply:
I was thinking more of trusting the client to behave itself. A "good" client will 
1. request the controller config, check that it's compatible
2. request a lock. If the lock registration fails, then it should fail to start. Or, the user can verify that the lock is stale and run some separate command to release the lock (or a flag to the client?)
3. request an unlock when it terminates

The main reason for trusting the client is that we might have clients that don't need to lock the controller but would still be requesting state changes (i.e. ghost birds for testing protocols, interfaces for monitoring the apparatus). And I'm not really worried about security, so adding an extra token seems like it's needless complexity.